grgit-gradle/src/main/groovy/org/ajoberstar/grgit/gradle/GrgitPlugin.groovy

```diff
@@ -22,11 +22,11 @@ class GrgitPlugin implements Plugin<Project> {
         grgit.close()
       }
 
-      project.allprojects { prj ->
-        if (prj.ext.has('grgit')) {
+      project.allprojects { Project prj ->
+        if (prj.extensions.hasProperty('grgit')) {
           prj.logger.warn("Project ${prj.path} already has a grgit property. Remove org.ajoberstar.grgit from either ${prj.path} or ${project.path}.")
         }
-        prj.ext.grgit = grgit
+        prj.extensions.add(Grgit, 'grgit', grgit)
       }
     } catch (Exception e) {
       project.logger.debug("Failed trying to find git repository for ${project.path}", e)
```

Comment:
You don't really want to keep the line below this down here.
`project.ext.grgit = null` should be removed.

Reply:
Unfortunately, I do want to keep it. Historically, there were some concerns about getting missing property exceptions when there wasn't a repo in the project. This way you can do some null safety checks, but it won't blow up just because you referenced it. Not what Is prefer, but there's a reason.